### DISCUSSION TITLE
Pro & Cons of Brave vs. Chrome

### DISCUSSION THREAD

**Discussion by alatteri** [dmarmor/epichrome#301] [Q&A]
Hi,

Wonder if @dmarmor could give a quick/short comparison of the pros and cons of using the two available engines.

Thanks,
Alan

**Comment by dmarmor** [comment]
I'm happy to! The tl;dr is that for most applications I much prefer the Brave engine, which offers a cleaner, more truly app-like experience in a few ways. Here's my breakdown:

**ADVANTAGES OF THE BUILT-IN BRAVE ENGINE:**

- If you want your app to send system notifications, those notifications will have your correct custom icon, whereas with Chrome-based apps, notifications always use the generic Chrome icon.

- If your app needs access to the camera or microphone, do screen recording, or use other system resources that require granting system permission, each app can be separately given whatever permissions it needs. If you use the Chrome engine, your apps can still access most resources but doing so will also give access to Chrome itself and every other Chrome-based Epichrome app on your system. (The exception to this is screen recording, which won’t work at all with Chrome-engine apps.)

- If you use scripting or the command line to interact with your apps and want to address them by ID, you can do that only with Brave apps. Chrome-based apps have unique IDs *only* when they are not running. when Chrome-based app runs, it adopts Chrome's ID and is indistinguishable from Chrome to anything that uses ID to find apps. (This is why system permissions can't be granted individually to Chrome-based apps.)

- Brave-based apps are likely to play nicer with URL-routing software like Choosy and Browser Fairy. Some of these don't like the fact that Chrome-based apps change ID when running, and won't work well with them.

- Chrome apps seem more prone to losing their icons and/or menubar names while running. I haven't been able to fully diagnose this, but I suspect it's something to do with the fact that they all have the same ID as Chrome while running, and the system can get confused.


**ADVANTAGES OF THE EXTERNAL CHROME ENGINE**

- Chrome apps run in an unaltered Chrome browser, meaning it is fully signed and thus compatible with certain extensions that Brave-based apps are not. The most common example of this is the old 1Password extension that communicates with the desktop app. This will not work on Brave-based apps, but does work on Chrome-based apps (as long as Epichrome has been installed under `/Applications`).

- You can use your Google account to sync extensions or other settings across multiple apps. (Note that Brave has been developing their own sync, but in my limited testing it's still pretty rough.)


That's it! Please reply in the comments if you've found other pros or cons to either engine, and I will add them to this list and keep it pinned as a reference.
